tests/tests_api/test_group.py

```diff
@@ -77,24 +101,180 @@ def test_public_group_retrieve(api_client):
+        for teacher in group.teachers.iterator():
+            assert teacher.status in (
+                TeacherStatus.TEACHING_ACCEPTING_MORE,
+                TeacherStatus.TEACHING_NOT_ACCEPTING_MORE,
+            )
+            assert teacher.status_since == common_status_since
+
+            log_event: TeacherLogEvent = TeacherLogEvent.objects.get(teacher_id=teacher.pk)
+            assert log_event.type == TeacherLogEventType.STUDY_START
+            self._compare_date_time_with_timestamp(log_event.date_time, timestamp)
+
+    @pytest.mark.parametrize(
+        "number_of_groups_to_start, expected_status",
+        [
+            (CoordinatorGroupLimit.MIN - 1, CoordinatorStatus.WORKING_BELOW_THRESHOLD),
+            (CoordinatorGroupLimit.MIN, CoordinatorStatus.WORKING_OK),
+            (CoordinatorGroupLimit.MIN + 1, CoordinatorStatus.WORKING_OK),
+            (CoordinatorGroupLimit.MAX - 1, CoordinatorStatus.WORKING_OK),
+            (CoordinatorGroupLimit.MAX, CoordinatorStatus.WORKING_LIMIT_REACHED),
+            (CoordinatorGroupLimit.MAX + 1, CoordinatorStatus.WORKING_LIMIT_REACHED),
+        ],
+    )
+    def test_public_group_start_coordinator_status(
+        self, api_client, timestamp, number_of_groups_to_start, expected_status
+    ):
+        coordinator = baker.make(Coordinator, _fill_optional=True)
+        coordinator.status = CoordinatorStatus.WORKING_BELOW_THRESHOLD
+        coordinator.save()
+
+        for i in range(number_of_groups_to_start):
+            group = baker.make(Group, _fill_optional=True)
+            group.status = GroupStatus.AWAITING_START
+            group.coordinators.add(coordinator)
+            group.save()
+
+            response = api_client.post(self._make_url(group))
+            assert response.status_code == status.HTTP_201_CREATED
+
+        coordinator.refresh_from_db()
+        assert coordinator.groups.count() == number_of_groups_to_start
+        assert coordinator.status == expected_status
+        self._compare_date_time_with_timestamp(coordinator.status_since, timestamp)
+
+    def test_public_group_start_student_status(self, api_client, timestamp):
+        student = baker.make(Student, _fill_optional=True)
+        student.status = StudentStatus.AWAITING_START
+        student.save()
+
+        group = baker.make(Group, _fill_optional=True)
+        group.status = GroupStatus.AWAITING_START
+
+        group.students.add(student)
+
+        group.save()
+
+        response = api_client.post(self._make_url(group))
+        assert response.status_code == status.HTTP_201_CREATED
+
+        student.refresh_from_db()
+        assert student.groups.count() == 1
+        assert student.status == StudentStatus.STUDYING
+        self._compare_date_time_with_timestamp(student.status_since, timestamp)
+
+    @staticmethod
+    def _compare_date_time_with_timestamp(
+        date_time: datetime.datetime, timestamp: datetime.datetime
+    ):
+        assert date_time.year == timestamp.year
```

Comment:
Why don't you compare objects directly?

Reply:
Because test `timestamp` that is generated in `conftest.py` will never be exactly the same as the timestamp generated in the group processor. That's why I don't even check the seconds (just in case). Furthermore, when log events are generated, current time is added to them automatically, as defined in `LogEvent`:
```py
date_time = models.DateTimeField(auto_now_add=True)
```
So direct comparison will always fail.